### PR TITLE
Ruby on Rails capstone project - Budget app

### PR DESCRIPTION
**Pull request by aron-net**
### Interactions

-  [X] Splash screen
      - [ ]  A simple page with the name of your app (yes, you need to choose one), and links to the sign up and log in pages.

 -  [X] Sign up and log in pages
       - [ ]  The user should be able to register in the app with full name, email and password (all mandatory).
       - [ ] The user can log into the app using email and password.
       - [ ] If the user is not logged in, they can't access pages that require the user to be logged in (all the pages described below).

- [X] Home page (catagories page)
     - [ ] When the user logs in, they are presented with the categories page.
     - [ ] For each category, the user can see their name, icon and the total amount of all the transactions that belongs to that category.
     - [ ]  When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
     - [ ] There is a button "add a new category" at the bottom that brings the user to the page to create a new category.

- [X] Transactions page
     - [ ]  For a given category, the list of transactions is presented, ordered by the most recent.
    - [ ]  At the top of the page the user could see the total amount for the category (sum of all of the amounts of the transactions in that category).
    - [ ]  There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.
     - [ ]   When the user clicks on the "Back" button (<), the user navigates to the home page.

- [X]  "Add a new category" page
     - [ ]  The user fills out a form to create a new category, indicating their name and icon (both mandatory).
     - [ ] The user click (or taps) the "Save" button to create the new category, and is taken to the home page on success.
     - [ ]   When the user clicks on the "Back" button (<), the user navigates to the home page.

 - [X] "Add a new transaction" page
      - [ ]  The user fills out a form to create a new transaction with:
           -  name (mandatory)
           - amount (mandatory)
           -  categories (mandatory at least one)
      - [ ] The user click (or taps) the "Save" button to create the new transation, and is taken to the transactions page for that category.
     - [ ] When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.

- [X] Video 
https://www.loom.com/share/88d73495f1644857938b7e01f1891a0b
- [X] Second video
- https://www.loom.com/share/6b8448befac84de59f3db442757f6e81

### CODE REVIEW

File: app/assets/stylesheets/category.css [hunk 1]
```diff
@@ -0,0 +1,92 @@
+.title {
+  background-color: #5fb523;
+  padding: 1rem;
+  text-align: center;
+  color: white;
+  position: relative;
+}
+
+#categories {
```
Comment (arslanbisharat) [inline app/assets/stylesheets/category.css]: - [ ] It is not recommended to use ID as a selector. You can use classes instead.

File: app/assets/stylesheets/category.css [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+  font-size: 1.5rem;
+}
+
+.transaction_total {
+  right: 3rem;
+  bottom: 1.5rem;
+  font-size: 1.5rem;
+}
+
+#icon_container {
```
Comment (arslanbisharat) [inline app/assets/stylesheets/category.css]: - [ ]  It is not recommended to use ID as a selector. You can use classes instead.

File: app/assets/stylesheets/category.css [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+  width: 30%;
+  font-size: 1.5em;
+  text-align: center;
+  outline: none;
+  border: 2px solid rgb(180, 156, 156);
+}
+
+#category-icon {
```
Comment (arslanbisharat) [inline app/assets/stylesheets/category.css]: - [ ]  It is not recommended to use ID as a selector. You can use classes instead.

File: app/models/category.rb [hunk 4]
```diff
@@ -0,0 +1,7 @@
+class Category < ApplicationRecord
+  belongs_to :user
+  has_and_belongs_to_many :payments
+
+  # validates :name, presence: true, length: { minimum: 3 }
+  # validates :user_id, presence: true, numericality: { greater_than: 0 }
```
Comment (arslanbisharat) [inline app/models/category.rb]: - [ ] Let's remove commented code from your project. This makes it hard to read it.

File: app/models/user.rb [hunk 5]
```diff
@@ -0,0 +1,8 @@
+class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
```
Comment (arslanbisharat) [inline app/models/user.rb]: - [ ] Let's remove commented code from your project. This makes it hard to read it.

File: README.md [hunk 6]
```diff
@@ -47,7 +47,7 @@ After you're finished please remove all the comments and instructions!
   - [Usage](#usage)
   - [Run tests](#run-tests)
```
Comment (arslanbisharat) [inline README.md:48]: - [ ] There is no sound in the audio recording. Please record a new video.

File: README.md [hunk 7]
```diff
@@ -47,7 +47,7 @@ After you're finished please remove all the comments and instructions!
...
   - [Deployment](#triangular_flag_on_post-deployment)
-- [👥 Authors](#authors)
+- [👥 Author](#author)
 - [🔭 Future Features](#future-features)
 - [🤝 Contributing](#contributing)
 - [⭐️ Show your support](#support)
```
Comment (arslanbisharat) [inline README.md:53]: - [ ] Please make sure the README contains an "Acknowledgements" section or similar that gives appropriate credit to [Gregoire Vella on Behance](https://www.behance.net/gregoirevella) the [author of the original design](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=), as required by the [Creative Commons license of the design](https://creativecommons.org/licenses/by-nc/4.0/).

**Review by arslanbisharat — CHANGES_REQUESTED** [review]
Hi @aron-helu, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

- The Pull Request has a proper title and description. ✔️
- You have added a descriptive Readme file. Good Job. ✔️
- There is no linter error is present. ✔️
- Project has met all the technical aspects you needed for this project. Keep rocking! ✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- [ ] Please make sure you right descriptive comments. Commenting involves placing Human Readable Descriptions inside of computer programs detailing what the Code is doing. Proper use of commenting can make code maintenance much easier, as well as helping make finding bugs faster. Further, commenting is very important when writing functions that other people will use. :x:


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/assets/stylesheets/category.css (hunk 1 above)
Comment (aron-net) [reply app/assets/stylesheets/category.css]: Implemented thank you for reviewing. :+1:

File: app/assets/stylesheets/category.css (hunk 2 above)
Comment (aron-net) [reply app/assets/stylesheets/category.css]: Implemented thank you for reviewing. :+1:

File: app/assets/stylesheets/category.css (hunk 3 above)
Comment (aron-net) [reply app/assets/stylesheets/category.css]: Implemented thank you for reviewing. :+1:

File: app/models/category.rb (hunk 4 above)
Comment (aron-net) [reply app/models/category.rb]: Implemented thank you for reviewing. :+1:

File: app/models/user.rb (hunk 5 above)
Comment (aron-net) [reply app/models/user.rb]: Implemented thank you for reviewing. :+1:

File: README.md (hunk 6 above)
Comment (aron-net) [reply README.md:48]: Implemented thank you for reviewing. :+1:

File: README.md (hunk 7 above)
Comment (aron-net) [reply README.md:53]: Implemented thank you for reviewing. :+1:

**Review by andersonlebon — APPROVED** [review]
### Hello @aron-helu 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 :clap: :clap: :100: 

nodding-yes

## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Good PR description :heavy_check_mark: 
- Your README file is now professional :heavy_check_mark: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------

**Pull request merged**